src/Proxy/ProxyFactory.php

```diff
@@ -195,319 +41,36 @@ public function __construct(
-            }
-
-            return;
-        }
-
-        $parentDirectory = dirname($fileName);
-
-        if (! is_dir($parentDirectory) && ! @mkdir($parentDirectory, 0775, true)) {
-            throw ORMInvalidArgumentException::proxyDirectoryNotWritable($this->proxyDir);
+        foreach ($identifier as $idField => $value) {
+            $classMetadata->propertyAccessors[$idField]->setValue($proxy, $value);
         }
 
-        if (! is_writable($parentDirectory)) {
-            throw ORMInvalidArgumentException::proxyDirectoryNotWritable($this->proxyDir);
-        }
-
-        $tmpFileName = $fileName . '.' . bin2hex(random_bytes(12));
-
-        file_put_contents($tmpFileName, $proxyCode);
-        @chmod($tmpFileName, 0664);
-        rename($tmpFileName, $fileName);
+        return $proxy;
     }
 
-    private function generateUseLazyGhostTrait(ClassMetadata $class): string
-    {
-        // @phpstan-ignore staticMethod.deprecated (Because we support Symfony < 7.3)
-        $code = ProxyHelper::generateLazyGhost($class->getReflectionClass());
-        $code = substr($code, 7 + (int) strpos($code, "\n{"));
-        $code = substr($code, 0, (int) strpos($code, "\n}"));
-        $code = str_replace('LazyGhostTrait;', str_replace("\n    ", "\n", 'LazyGhostTrait {
-            initializeLazyObject as private;
-            setLazyObjectAsInitialized as public __setInitialized;
-            isLazyObjectInitialized as private;
-            createLazyGhost as private;
-            resetLazyObject as private;
-        }
-
-        public function __load(): void
-        {
-            $this->initializeLazyObject();
-        }
-        '), $code);
-
-        return $code;
-    }
-
-    private function generateSerializeImpl(ClassMetadata $class): string
-    {
-        $reflector  = $class->getReflectionClass();
-        $properties = $reflector->hasMethod('__serialize') ? 'parent::__serialize()' : '(array) $this';
-
-        $code = '$properties = ' . $properties . ';
-        unset($properties["\0" . self::class . "\0lazyObjectState"]);
-
-        ';
-
-        if ($reflector->hasMethod('__serialize') || ! $reflector->hasMethod('__sleep')) {
-            return $code . 'return $properties;';
-        }
-
-        return $code . '$data = [];
-
-        foreach (parent::__sleep() as $name) {
-            $value = $properties[$k = $name] ?? $properties[$k = "\0*\0$name"] ?? $properties[$k = "\0' . $reflector->name . '\0$name"] ?? $k = null;
-
-            if (null === $k) {
-                trigger_error(sprintf(\'serialize(): "%s" returned as member variable from __sleep() but does not exist\', $name), \E_USER_NOTICE);
-            } else {
-                $data[$k] = $value;
-            }
-        }
-
-        return $data;';
-    }
-
-    private static function generateProxyClassName(string $className, string $proxyNamespace): string
+    protected function skipClass(ClassMetadata $metadata): bool
```

Comment:
This looks unused in this context: if it is in use, let's indeed hit it with a couple tests

Reply:
Removed, it indeed wasn't in use.